src/tactic/suggest.lean

```diff
@@ -148,6 +148,21 @@ meta def unpack_iff_both : list decl_data → list decl_data
 | (⟨d, n, both, l⟩ :: L) := ⟨d, n, mp, l⟩ :: ⟨d, n, mpr, l⟩ :: unpack_iff_both L
 | (⟨d, n, m, l⟩ :: L)    := ⟨d, n, m, l⟩ :: unpack_iff_both L
 
+/-- An extension to the option structure for `solve_by_elim`,
+to specify a list of local hypotheses which must appear in any solution.
+These are useful for constraining the results from `library_search` and `suggest`. -/
+meta structure suggest_opt extends opt :=
+(compulsory_hyps : list expr := [])
+
+/--
+Convert a `suggest_opt` structure to a `opt` structure suitable for `solve_by_elim`,
+by setting the `accept` parameter to require that all complete solutions
+use everything in `compulsory_hyps`.
+-/
+meta def suggest_opt.mk_accept (o : suggest_opt) : opt :=
+{ accept := λ gs, guard $ o.compulsory_hyps.all (λ h, gs.any (λ g, g.contains_expr_or_mvar h)),
```

Comment:
Is it a problem that this overwrites the old `accept`? Will you ever have a `suggest_opt` with a non-default `accept`?

Reply:
Good point. I worried momentarily about this then decided to be lazy. I'll fix it now to just check the old `accept` first.

Reply:
Done.